kafka-commons/src/main/java/uk/gov/homeoffice/digital/sas/kafka/consumer/KafkaConsumerUtils.java

```diff
@@ -0,0 +1,37 @@
+package uk.gov.homeoffice.digital.sas.kafka.consumer;
+
+import static uk.gov.homeoffice.digital.sas.kafka.constants.Constants.KAFKA_DESERIALIZATION_TO_CONCRETE_TYPE_FAILED;
+import static uk.gov.homeoffice.digital.sas.kafka.constants.Constants.KAFKA_SUCCESSFUL_DESERIALIZATION;
+import static uk.gov.homeoffice.digital.sas.kafka.constants.Constants.RESOURCE_JSON_ATTRIBUTE;
+import static uk.gov.homeoffice.digital.sas.kafka.constants.Constants.SCHEMA_JSON_ATTRIBUTE;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import org.springframework.util.ObjectUtils;
+import uk.gov.homeoffice.digital.sas.kafka.message.Messageable;
+
+@Slf4j
+@Component
+public class KafkaConsumerUtils<T extends Messageable> {
+
+  public void checkDeserializedResource(String payload, T resource) {
```

Comment:
This method would fit nicely in the type KafkaConsumerService.
Then the current class would remain a pure utility class accessed staticall

Reply:
This method was moved to the kafka consumer service.